### PR TITLE
Add the morphology stack

### PR DESCRIPTION
**Pull request by wizmer**

### CODE REVIEW

**Review by matz-e — APPROVED** [review]
LGTM

**Comment by wizmer** [comment]
Can we retrigger this build ?
The current error is:
`17:28:54  IOError: [Errno 26] Text file busy: '/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/663/spack/venv/bin/python'` which might be a temporary issue.

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
I deleted the PR directory… should fix this

**Comment by pramodk** [comment]
I see : 

```
[2020-01-30T13:10:06.986Z] 
[2020-01-30T13:10:06.986Z] py-scikit-learn requires python version 3.5:, but spec asked for 2.7.15
```
So newer scikit-learn requires Python3. So either we should install old version if we need python2 Or we should [scikit-lear from](https://github.com/BlueBrain/spack/blob/ea8b8a24795140958b0a7e5d6b4092b2ada60c8d/deploy/packages/python-packages.yaml#L157) here and should add new python3 only block:

```
  gnu-stable-serial-python3-lapack:
    target_matrix:
      - gnu-stable
    requires:
      - architecture
      - compiler
      - lapack
      - python
    specs:
      - py-scikit-learn
```

File: var/spack/repos/builtin/packages/morphio/package.py [hunk 1]
```diff
@@ -27,7 +27,8 @@ class Morphio(CMakePackage):
     depends_on('mpi', when='+mpi')
 
     def cmake_args(self):
-        args = ['-DBUILD_BINDINGS:BOOL=OFF']
+        args = ['-DBUILD_BINDINGS:BOOL=ON']
+        args = []
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/morphio/package.py]: What?

File: var/spack/repos/builtin/packages/py-bbp-morphology-workflow/package.py [hunk 2]
```diff
@@ -17,6 +17,7 @@ class PyBbpMorphologyWorkflow(PythonPackage):
     depends_on('py-setuptools', type=('build', 'run'))
 
     depends_on('py-neuroc', type='run')
+    depends_on('py-scipy', type='run')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-bbp-morphology-workflow/package.py]: `py-morphtool` already depends on this. Is it a direct dependency?

File: var/spack/repos/builtin/packages/py-morphio/package.py [hunk 3]
```diff
@@ -10,9 +10,11 @@ class PyMorphio(PythonPackage):
     """Python library for reading / writing morphology files"""
 
     homepage = "https://github.com/BlueBrain/MorphIO/"
+    url      = "https://github.com/BlueBrain/MorphIO.git"
     git      = "https://github.com/BlueBrain/MorphIO.git"
 
-    version('develop', branch='master', submodules=True, clean=False)
+    version('develop', git=url, submodules=True, clean=False)
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-morphio/package.py]: Why this change?

File: var/spack/repos/builtin/packages/py-bbp-morphology-workflow/package.py (hunk 2 above)
Comment (wizmer) [reply var/spack/repos/builtin/packages/py-bbp-morphology-workflow/package.py]: Because it was still complaining that scipy was not found even though it was in the morph-tool recipe. But I think was Spack env was screwed.

File: var/spack/repos/builtin/packages/py-cut-plane/package.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-cut-plane/package.py]: Should be 2020 by now :)

File: var/spack/repos/builtin/packages/py-neurom/package.py [hunk 5]
```diff
@@ -10,8 +10,12 @@ class PyNeurom(PythonPackage):
     """Python library neuron morphology analysis"""
 
     homepage = "https://github.com/BlueBrain/NeuroM"
+    git = "https://github.com/BlueBrain/NeuroM.git"
     url = "https://pypi.io/packages/source/n/neurom/neurom-1.4.10.tar.gz"
 
+    version('develop', branch='master')
+    version('mut_morphio', branch='mut_morphio')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-neurom/package.py:17]: I'm a little uneasy here, since this is a moving target. I hope there's a plan to merge this at some point?

File: var/spack/repos/builtin/packages/py-pyquaternion/package.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-pyquaternion/package.py]: 2020

**Review by matz-e — DISMISSED** [review]
Looks OK to me. Some non-blocking comments… also, do you plan to feed some of the new packages upstream? I've been doing that manually every now and then, but spreading that load will help with keeping us from diverging with upstream too much.

File: var/spack/repos/builtin/packages/py-neurom/package.py (hunk 5 above)
Comment (wizmer) [reply var/spack/repos/builtin/packages/py-neurom/package.py:17]: I'm uneasy as well. Yes there is a plan to merge this.

**Comment by wizmer** [comment]
Yes, I was planning to propagate `py-quaternion` upstream. The others are bbp internals so this is the only one.

**Comment by matz-e** [comment]
Retest this please

File: var/spack/repos/builtin/packages/py-neuroc/package.py [hunk 7]
```diff
@@ -10,6 +10,7 @@ class PyNeuroc(PythonPackage):
     """Python library neuron morphology analysis"""
 
     homepage = "https://bbpcode.epfl.ch/code/#/admin/projects/nse/neuroc"
+    url = "https://bbpcode.epfl.ch/code/#/admin/projects/nse/neuroc"
     git      = "ssh://bbpcode.epfl.ch/nse/NeuroC"
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-neuroc/package.py]:
```suggestion
    git = "ssh://bbpcode.epfl.ch/nse/NeuroC"
```

**Comment by wizmer** [comment]
I don't understand why the build is failing with a complain on the NeuroM version.
On my local Spack I can build all packages ? And the output of `spack spec py-morph-tool` only shows a dependency on Neurom@mut_morphio. I don't understand who is bringing Neurom@1.4.15 in the mix ?
```
coste@bbpv1:~/workspace/spack$ spack spec py-morph-tool
Input spec
--------------------------------
py-morph-tool

Concretized
--------------------------------
py-morph-tool@0.1.15%gcc@8.3.0~neuron arch=linux-rhel7-x86_64
    ^py-click@7.0%gcc@8.3.0 arch=linux-rhel7-x86_64
        ^py-setuptools@41.4.0%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^python@3.7.4%gcc@8.3.0+bz2+ctypes+dbm+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4~uuid+zlib arch=linux-rhel7-x86_64
    ^py-neurom@mut_morphio%gcc@8.3.0+plotly arch=linux-rhel7-x86_64
        ^py-future@0.17.1%gcc@8.3.0 arch=linux-rhel7-x86_64
        ^py-h5py@2.10.0%gcc@8.3.0~mpi arch=linux-rhel7-x86_64
            ^hdf5@1.10.5%gcc@8.3.0~cxx~debug~fortran+hl~mpi+pic+shared~szip~threadsafe arch=linux-rhel7-x86_64
                ^zlib@1.2.11%gcc@8.3.0+optimize+pic+shared arch=linux-rhel7-x86_64
            ^py-cached-property@1.5.1%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^py-cython@0.29.13%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^py-numpy@1.17.3%gcc@8.3.0+blas+lapack arch=linux-rhel7-x86_64
                ^intel-mkl@2019.3.199%gcc@8.3.0~ilp64+shared threads=none arch=linux-rhel7-x86_64
            ^py-pkgconfig@1.4.0%gcc@8.3.0 arch=linux-rhel7-x86_64
                ^pkg-config@0.29.2%gcc@8.3.0+internal_glib arch=linux-rhel7-x86_64
                ^py-nose@1.3.7%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^py-six@1.12.0%gcc@8.3.0 arch=linux-rhel7-x86_64
        ^py-matplotlib@3.1.1%gcc@8.3.0~animation backend=agg +image~latex~movies arch=linux-rhel7-x86_64
            ^freetype@2.10.1%gcc@8.3.0 arch=linux-rhel7-x86_64
                ^bzip2@1.0.6%gcc@8.3.0+shared arch=linux-rhel7-x86_64
                ^libpng@1.6.37%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^py-cycler@0.10.0%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^py-kiwisolver@1.1.0%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^py-pillow@6.2.0%gcc@8.3.0~freetype~jpeg2000~lcms~tiff arch=linux-rhel7-x86_64
                ^binutils@2.31.1%gcc@8.3.0+gold~headers~libiberty+nls~plugins arch=linux-rhel7-x86_64
                ^libjpeg-turbo@2.0.3%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^py-pyparsing@2.4.2%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^py-python-dateutil@2.8.0%gcc@8.3.0 arch=linux-rhel7-x86_64
                ^py-setuptools-scm@3.3.3%gcc@8.3.0 arch=linux-rhel7-x86_64
        ^py-morphio@2.3.4%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^cmake@3.15.3%gcc@8.3.0~doc+ncurses+openssl+ownlibs patches=1e1e5303be75213d46a2da90bcd977b8ac5c4fcb807802b1f4eb29d5449e5dce ~qt arch=linux-rhel7-x86_64
        ^py-plotly@3.0.0%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^py-pytz@2019.3%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^py-requests@2.22.0%gcc@8.3.0 arch=linux-rhel7-x86_64
                ^py-certifi@2019.9.11%gcc@8.3.0 arch=linux-rhel7-x86_64
                ^py-chardet@3.0.4%gcc@8.3.0 arch=linux-rhel7-x86_64
                    ^py-pytest-runner@5.1%gcc@8.3.0 arch=linux-rhel7-x86_64
                ^py-idna@2.8%gcc@8.3.0 arch=linux-rhel7-x86_64
                ^py-urllib3@1.25.6%gcc@8.3.0 arch=linux-rhel7-x86_64
        ^py-pylru@1.1.0%gcc@8.3.0 arch=linux-rhel7-x86_64
        ^py-pyyaml@5.1.2%gcc@8.3.0 arch=linux-rhel7-x86_64
            ^libyaml@0.2.2%gcc@8.3.0 arch=linux-rhel7-x86_64
        ^py-scipy@1.3.1%gcc@8.3.0 arch=linux-rhel7-x86_64
        ^py-tqdm@4.36.1%gcc@8.3.0 arch=linux-rhel7-x86_64
    ^py-pathlib2@2.3.5%gcc@8.3.0 arch=linux-rhel7-x86_64
```

**Comment by pramodk** [comment]
Did you try FAQ “How do I debug my Pull Request” ? May be that will help to reproduce the issue locally?

**Comment by wizmer** [comment]
Thanks, I was not aware of it. I am trying this right now !

**Comment by wizmer** [comment]
I don't manage to make this PR pass. I will break it down into smaller PRs.

**Comment by wizmer** [comment]
Abandonned as this has been merged in https://github.com/BlueBrain/spack/pull/722 instead
